spatialpandas/geometry/base.py

```diff
@@ -0,0 +1,640 @@
+            arrow_dtype = pa.from_numpy_dtype(dtype)
+
+            # Wrap dtype with appropriate number of nesting levels
+            for i in range(self._nesting_levels):
+                arrow_dtype = pa.list_(arrow_dtype)
+        else:
+            # Let arrow infer type
+            arrow_dtype = None
+
+        # Unwrap Geometry elements to numpy arrays
+        if is_array_like(array) or isinstance(array, list):
+            array = [_unwrap_geometry(el, self._element_type) for el in array]
+            self.data = pa.array(array, type=arrow_dtype)
+        elif isinstance(array, pa.Array):
+            self.data = array
+        elif isinstance(array, pa.ChunkedArray):
+            self.data = pa.concat_arrays(array)
+        else:
+            raise ValueError(
+                "Unsupported type passed for {}: {}".format(
+                    self.__class__.__name__, type(array)
+                )
+            )
+
+        self.offsets = np.array([0])
+
+        # Check that inferred type has the right number of nested levels
+        pyarrow_element_type = _validate_nested_arrow_type(
+            self._nesting_levels, self.data.type
+        )
+
+        self._pyarrow_element_type = pyarrow_element_type
+        self._numpy_element_type = pyarrow_element_type.to_pandas_dtype()
+        self._dtype = self._dtype_class(self._numpy_element_type)
+
+        # Validate input data is compatible
+        offsets = self.buffer_offsets
+
+        # Validate even number of inner elements per polygon
+        if any((offsets[-1] % 2) > 0):
+            raise ValueError("""
+Geometry objects are represented by interleaved x and y coordinates, so they must have
+an even number of elements. Received specification with an odd number of elements.""")
+
+    @property
+    def _dtype_class(self):
+        return GeometryDtype
+
+    @property
+    def dtype(self):
+        return self._dtype
+
+    @property
+    def nbytes(self):
+        size = 0
+        for buf in self.data.buffers():
+            if buf is not None:
+                size += buf.size
+        return size
+
+    nbytes.__doc__ = ExtensionArray.nbytes.__doc__
+
+    def astype(self, dtype, copy=True):
+        if self.dtype == dtype:
+            return self
+
+        if dtype == np.dtype('object'):
+            return np.array(self, dtype='object')
+
+        if isinstance(dtype, GeometryDtype):
+            dtype = dtype.arrow_dtype.to_pandas_dtype()
+        elif isinstance(dtype, pa.DataType):
+            dtype = dtype.to_pandas_dtype()
+        else:
+            dtype = np.dtype(dtype)
+
+        return GeometryArray(np.asarray(self).astype(dtype), dtype=dtype)
+
+    astype.__doc__ = ExtensionArray.astype.__doc__
+
+    def isna(self):
+        return extract_isnull_bytemap(self.data)
+
+    isna.__doc__ = ExtensionArray.isna.__doc__
+
+    def copy(self, deep=False):
+        if deep:
+            raise NotImplementedError("Deep copy is not supported")
```

Comment:
What should people do if they do need a deep copy, e.g. so they can edit it? Is there a suggestion we could make here?